docs/SONATA_DEVELOPER_GUIDE.md

```diff
@@ -904,107 +904,98 @@ The indexing is rooted at a single group called indices which is a subgroup of a
+The source_to_target/range_to_edge_id dataset defines ranges of edges in the edge population where each range contains edges whose source node is the same (the target node should also be the same for compactness). This dataset is meant to be indexed with the indices from node_id_to_range. For each row the two values define the start index of the range and the non-inclusive end index.
 
-### <a name="neuron_config">Tying it all together - the network/circuit config file
+The datasets from the target_to_source group are defined symmetrically. From this symmetry is easy to infer that edges should be grouped by source, target pairs, otherwise an important overhead will be incurred in one or both of the indices.
 
-The config file is a .json file that defines the relative location of each part of the network:
-
-{
+### <a name="network_config">Tying it all together - the network/circuit config file
 
-    "target_simulator":"NEURON",
+The config file is a JSON file that defines the relative location of each part of the network.
 
-target_simulator can be "NEURON", “PyNN”, “NEST”, etc.  It specifies the intended target simulator of the circuit description.  For now, this field is intended as a declaration, and an implementation may decide to throw an error for unsupported targets.  In practice, mechanism and parameter names are tailored to the given target simulator.  For model_type=biophysical NEURON is supported, but not PyNN or NEST.
+It includes the following sections / values:
 
-The "manifest" section of the config file provides a convenient handle on setting variables that point to base paths.  These variables can be then used in the rest of the config file to point to various directories that share the first portion of the path.
+#### target_simulator
 
-    "manifest": {
+Can be "NEURON", "PyNN", "NEST", etc. It specifies the intended target simulator of the circuit description. For now, this field is intended as a declaration, and an implementation may decide to throw an error for unsupported targets. In practice, mechanism and parameter names are tailored to the given target simulator. For `model_type`=`biophysical` NEURON is supported, but not PyNN or NEST.
 
-        "$BASE_DIR": "/path/to/my/workspace",
-
-        "$NETWORK_DIR": "$BASE_DIR/networks",
-
-        "$COMPONENT_DIR": "$BASE_DIR/components"
-
-    },
+Example:
+```
+"target_simulator": "NEURON"
+```
 
-    "components_dir": {
+#### manifest
 
-The directory in which to find the neuronal morphology files:
+Provides a convenient handle on setting variables that point to base paths. These variables can be then used in the rest of the config file to point to various directories that share the first portion of the path. One can also use '.' in the beginning of any path to declare a path relevant to the network config file location.
```

Comment:
I think it's even easier to say: Paths are always relative to the directory containing the network config file unless there is a leading `/`, in which case they are considered as absolute paths. This way we could omit the base dir and say: 

    "$NETWORK_DIR": "networks"
What happens to Windows vs UNIX filesystem conventions? We haven't discussed that. Is "C:\foo\bar" a valid path. Should conforming implementation by able to understand that? This needs to be specified. Could you provide your view on this @kaeldai?

Reply:
The trickiest part of Windows paths in not the difference between \ and / but the C:\ style syntax. To be safe it seems better to keep the text that @arsenius7 suggested for relative paths.